### DISCUSSION TITLE
Enabling the use of an "if" statement right after the "return".

### DISCUSSION THREAD

**Discussion by Abab-bk** [dotnet/csharplang#8197] [Language Ideas] [answered]
Enabling the use of an "if" statement right after the "return".

For example:
```return if obj is null;```
This line of code does the same thing as:
```if (obj is null) return;```

In fact, it is exactly the same as ```if (obj is null) return;```. Removing the parentheses from this line of code achieves the same result. However, in order to maintain the style and compatibility of C#, we place the return at the beginning.

**Compatibility with previous versions:**
Your existing code will work as usual without requiring any changes.

**Comment by KirisameY** [comment]
thats a good idea that may change the rest of my life from God·Hua 🎉🎉🎉
but i think a return value is necessary, so how about:
`return [value] if (some_condition)`

**Comment by CyrusNajmabadi** [comment, marked as answer]
Dupe of  https://github.com/dotnet/csharplang/discussions/6345.
